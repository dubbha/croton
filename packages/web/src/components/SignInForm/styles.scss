.btn {
  width: 100%;
}

form {
  width: 70%;

  .register {
    margin-top: 16px;
  }
}

form.signinForm {
  margin-bottom: 20px;
  padding-top: 20px;
}
