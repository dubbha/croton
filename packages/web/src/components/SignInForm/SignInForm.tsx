import React, { useState, FormEvent, useEffect } from 'react';
import { Form, SubmitButton } from 'elements';
import { Link } from 'react-router-dom';
import './styles.scss';

type Props = {
  onSubmit: (email: string, password: string) => void;
};

export const SignInForm = ({ onSubmit }: Props) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [isValid, setIsValid] = useState(false);

  useEffect(() => {
    setIsValid(!!email && !!password);
  }, [email, password]);

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    onSubmit(email, password);
  };

  return (
    <>
      <Form onSubmit={handleSubmit}>
        <Form.Group controlId="formEmail">
          <Form.Label>Email address</Form.Label>
          <Form.Control
            type="email"
            placeholder="Enter email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            data-testid="signInForm__email"
          />
        </Form.Group>

        <Form.Group controlId="formPassword">
          <Form.Label>Password</Form.Label>
          <Form.Control
            type="password"
            placeholder="Enter password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            data-testid="signInForm__password"
          />
          <Form.Text className="reset-password-link">
            <Link to="/reset">Forgot password?</Link>
          </Form.Text>
        </Form.Group>
        <SubmitButton disabled={!isValid}>Sign Up</SubmitButton>
      </Form>
    </>
  );
};
