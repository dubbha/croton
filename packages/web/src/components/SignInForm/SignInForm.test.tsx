import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SignInForm } from './SignInForm';

describe('components/SignInForm', () => {
  const fn = jest.fn();

  it('should render successfully', () => {
    const { container } = render(
      <SignInForm onSubmit={fn} isLoading={false} error={null} />,
      { wrapper: MemoryRouter },
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render submit button disabled by default', () => {
    const { container, getByTestId } = render(
      <SignInForm onSubmit={fn} isLoading={false} error={null} />,
      { wrapper: MemoryRouter },
    );
    expect(container.firstChild).toMatchSnapshot();

    const submitButton = getByTestId('submitButton');
    expect(submitButton).toBeDisabled();
  });

  it('should call onSubmit prop on submit', () => {
    const { container, getByTestId } = render(
      <SignInForm onSubmit={fn} isLoading={false} error={null} />,
      { wrapper: MemoryRouter },
    );
    expect(container.firstChild).toMatchSnapshot();

    const emailInput = getByTestId('signInForm__email');
    const passwordInput = getByTestId('signInForm__password');
    const submitButton = getByTestId('submitButton');

    fireEvent.change(emailInput, { target: { value: 'admin@admin.com' } });
    fireEvent.change(passwordInput, { target: { value: 'admin' } });

    expect(submitButton).toBeEnabled();

    fireEvent.click(submitButton);
    expect(fn).toBeCalled();
  });

  it('should display error alert on error', () => {
    const error = 'ERROR';
    const { getByTestId } = render(
      <SignInForm onSubmit={fn} isLoading={false} error={error} />,
      { wrapper: MemoryRouter },
    );

    const errorAlert = getByTestId('errorAlert');

    expect(errorAlert.textContent).toBe('ERROR');
  });
});
