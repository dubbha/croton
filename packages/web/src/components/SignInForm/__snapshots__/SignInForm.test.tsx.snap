// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SignInForm should call onSubmit prop on submit 1`] = `
<form
  class=""
>
  <input
    class="form-control"
    data-testid="signInForm__email"
    placeholder="Enter email"
    type="email"
    value=""
  />
  <input
    class="form-control"
    data-testid="signInForm__password"
    placeholder="Enter password"
    type="password"
    value=""
  />
  <button
    class="btn btn-primary"
    data-testid="submitButton"
    disabled=""
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`components/SignInForm should render successfully 1`] = `
<form
  class=""
>
  <input
    class="form-control"
    data-testid="signInForm__email"
    placeholder="Enter email"
    type="email"
    value=""
  />
  <input
    class="form-control"
    data-testid="signInForm__password"
    placeholder="Enter password"
    type="password"
    value=""
  />
  <button
    class="btn btn-primary"
    data-testid="submitButton"
    disabled=""
    type="submit"
  >
    Submit
  </button>
</form>
`;
